core/src/test/java/org/elasticsearch/index/query/BaseQueryTestCase.java

```diff
@@ -125,7 +135,7 @@ protected void configure() {
         queryParserService = injector.getInstance(IndexQueryParserService.class);
         MapperService mapperService = queryParserService.mapperService;
         //create some random type with some default field, those types will stick around for all of the subclasses
-        currentTypes = new String[randomIntBetween(0, 5)];
+        currentTypes = new String[randomIntBetween(1, 5)];
```

Comment:
It kinda makes sense to me that the field is not there if we have no types (aka the index is empty). This is exactly why I want to keep testing against no types. I think this is a bug introduced with #11559 where a null check was removed (https://github.com/elastic/elasticsearch/commit/1fdae75025ebfa291436aadea4621671268daf82#commitcomment-11609013). The FieldNamesFieldMapper can be null and we should handle such situation properly. Once that gets fixed upstream our tests can still use no types randomly I think.